### ISSUE TITLE
hwloc-dump-hwdata.service installed in wrong location

### ISSUE THREAD

**Issue by opoplawski** [open-mpi/hwloc#221]
hwloc-dump-hwdata.service is installed into /usr/share/htwloc/ instead of /usr/lib/systemd/system.

Also, if this is really only needed on knight-landing systems - I probably should only be installed then, or at the very least be able to be disabled in configure.

**Comment by bgoglin** [comment]
Can you be sure that this /usr/lib/systemd/system is the correct location for all existing systemd installs out there? I tend to expect breakages depending on distros and system versions.

Even if we make install on non-KNL, the same install could be used on KNL if the file-system is shared. But you have to enable it in systemd if I remember correctly anyway so it shouldn't cause any problem anyway. And the tool also exits immediately if not running on KNL.
Hopefully, this won't matter in the near future because this tool will be used for non-KNL hosts :)

**Comment by opoplawski** [comment]
Yes, that location is standard among all systemd distributions that I'm aware of.

I'd be surprised to find /usr/lib shared between non-KNL and KNL machines.

I was wondering if this was starting to become a more general utility as well.. :)

**Comment by bgoglin** [comment]
I took a look at 16 packages that install a systemd service on my machine:
* 6 of them don't even provide the service upstream, but the packagers added one.
* 3 of them provide one and don't install it, the packagers install it.
* 7 others provide a service file and install it. 6 of them use pkg-config --variable=systemdsystemunitdir systemd to find the correct location. But most of them still have a configure option to change that path. Also several of these only install if systemd if explicitly enabled on the command-line, or if systemd if recent.
So it looks there's a proper solution emerging but it didn't convinced everybody yet. I am going to wait a bit before adding the required configury.

We'll also need to decide what to do with $prefix. Do we always install in /lib/systemd even if --with-prefix=/foo/bar is given? Or only if prefix is / or /usr? If we overwrite what a distribution package service file in /lib/systemd, we have to make sure the output of the new hwloc service is compatible with the hwloc tools installed by the package.

Regarding the shared /usr/lib, yes, it's not shared. But even if you install the service file in non-shared /lib, the hwloc-dump-hwdata program might still be in a shared file-system (when you have modules etc). If we stop installing the service (which isn't enabled by default anyway), we'll likely disable the program too.

hwloc-dump-hwdata is expected to dump all root-only data in the future (memory DIMM info in SMBIOS, PCI link speed, ...) but there are way more important things to do for the 2.0 release :/
